### PR TITLE
fix(DTFS2-7225): added API key check for missing endpoints

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
We currently have an API key check on Portal API endpoints that don’t have authentication checks, so as to prevent these endpoints being called from outside of DTFS. However, this API key check is missing from some endpoints that should have it:

* /login
* /users/reset-password
* /users/reset-password/:resetPwdToken

We should add the API key check to these endpoints to protect them from being called from outside of DTFS (which remains virtually impossible due to private links in Azure)

## Resolution :heavy_check_mark:
* Added `checkApiKey` to the endpoints.
* Created `HEADERS` constants in `libs/common` to adhere to `DRY` principle.

## Miscellaneous ➕ 
* Added missing `--workspaces` flag to `install:all` root package.json.

### CODE REVIEW

**Review by AlexBramhill — COMMENTED** [review]
One change if you think its needed :)  (also it seems the tests arent happy)

File: portal/server/api.test.js [hunk 1]
```diff
@@ -4,7 +4,12 @@ const api = require('./api');
 const { LOGIN_STATUS } = require('./constants');
 
 jest.mock('axios');
-const { PORTAL_API_URL } = process.env;
+const { PORTAL_API_URL, PORTAL_API_KEY } = process.env;
+
+const headers = {
+  'Content-Type': 'application/json',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal/server/api.test.js]: Have moved all to constant.

File: package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "db:seed:reset": "(cd utils && npm run db:seed:reset)",
     "env:copy": "(cd utils && npm run env:copy)",
     "housekeeping": "npm run update:all && npm run install:all && npm run prettier:fix:all && npm run lint:fix:all && npm run validate:yml:all && npm run audit:fix:all && npm run spellcheck",
-    "install:all": "npm install --if-present",
+    "install:all": "npm install --workspaces --if-present",
```
Comment (AlexBramhill) [inline package.json:36]: Not sure this is needed - https://github.com/UK-Export-Finance/dtfs2/blob/main/doc/npm-workspaces.md#notes-on-workspaces-flags

File: e2e-tests/ukef/cypress/support/portal-api/api.js [hunk 3]
```diff
@@ -1,10 +1,16 @@
+const { HEADERS } = require('@ukef/dtfs2-common');
 const { SIGN_IN_TOKEN_LINK_TOKEN } = require('../../../../../portal-api/api-tests/fixtures/sign-in-token-constants');
 
 const api = () => {
   const url = `${Cypress.config('dealApiProtocol')}${Cypress.config('dealApiHost')}:${Cypress.config('dealApiPort')}`;
   return url;
 };
 
+const headers = {
+  'x-api-key': Cypress.config('portalApiKey'),
```
Comment (AlexBramhill) [inline e2e-tests/ukef/cypress/support/portal-api/api.js:10]: I'm kind of surprised this worked without this being set as `test` before?

**Review by AlexBramhill — COMMENTED** [review]
lgtm after the 2 remaining comments, really nice job

**Review by AlexBramhill — APPROVED** [review]

**Review by AlexBramhill — DISMISSED** [review]

File: mongodb/Dockerfile [hunk 4]
```diff
@@ -1,8 +1,10 @@
+# Base image
 FROM mongo:5.0.5
 
 # Create keyfile
 RUN openssl rand -base64 756 > /mongo-keyfile
 
-# Set permissions and ownership for keyfile
-RUN chmod 400 /mongo-keyfile && \
-    chown 999:999 /mongo-keyfile
+# Set Read-only persmission to User
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply mongodb/Dockerfile:7]: Good spot, have rectified.

**Review by AlexBramhill — APPROVED** [review]

**Pull request merged**